### ISSUE TITLE
Lint/ShadowingOuterLocalVariable doesn't recognize when local variable is in another context in `case`

### ISSUE THREAD

**Issue by cheerfulstoic** [bbatsov/rubocop#1544]
I get this as an violation, but I don't think it should be:

```ruby
      case columns.size
      when 0
        fail ArgumentError, 'No columns specified for Query#pluck'
      when 1
        column = columns[0]
        query.map { |row| row[column] }
      else
        query.map do |row|
          columns.map do |column|
            row[column]
          end
        end
      end
```

The violation is for the `columns.map do |column|` in the `else`, but the `column` variable is in the other branch of the `case`.  Valid?

**Comment by jonas054** [comment]
This is a duplicate of #1518 because this is another situation in which `ruby` and `rubocop` report the same problem.

```
/tmp$ ruby -wc shadow2.rb 
shadow2.rb:9: warning: shadowing outer local variable - column
Syntax OK
/tmp$ rubocop -f s shadow2.rb 
== shadow2.rb ==
W:  9: 21: Shadowing outer local variable - column.

1 file inspected, 1 offense detected
```

**Comment by cheerfulstoic** [comment]
Cool, thanks for rubocop!
